astropy/io/misc/asdf/tags/transform/basic.py

```diff
@@ -90,6 +96,16 @@ def _to_tree_base_transform_members(cls, model, node, ctx):
             if bounds_nondefaults:
                 node['bounds'] = bounds_nondefaults
 
+        if not isinstance(model, CompoundModel):
+            if model.input_units is not None:
+                if model.input_units_equivalencies:
+                    input_unit_equivalencies = {}
```

Comment:
Is there a reason to rebuild the dict here?  Can we just set `node['input_units_equivalencies'] = model.input_units_equivalencies`?

Reply:
See changes in 97350e0.